tests/Feature/Shared/OfferControllerTest.php

```diff
@@ -0,0 +1,350 @@
+    public function user_cannot_update_others_users_offers(): void
+    {
+        $user = User::factory()->create();
+        $this->actingAs($user);
+
+        $offer = [
+            'url' => 'https://allegro.pl/',
+            'name' => 'New offer',
+            'is_active' => 1
+        ];
+
+        $response = $this->postJson(route('offers.store'), $offer)
+            ->assertCreated()
+            ->assertJsonFragment(['url' => 'https://allegro.pl/']);
+
+        $offerId = $response->json('data.id');
+
+        $user2 = User::factory()->create();
+        $this->actingAs($user2);
+
+        $updateResposne = $this->putJson(route('offers.update', $offerId), [
+            'is_active' => 0
+        ])
+            ->assertStatus(404);
+
+        $this->assertDatabaseHas('offers', $offer);
+    }
+
+    /** @test */
+    public function user_can_delete_his_offer(): void
+    {
+        $user = User::factory()->create();
+        $this->actingAs($user);
+
+        $offer = [
+            'url' => 'https://allegro.pl/',
+            'name' => 'New offer',
+            'is_active' => 1
+        ];
+
+        $response = $this->postJson(route('offers.store'), $offer)
+            ->assertCreated()
+            ->assertJsonFragment(['url' => 'https://allegro.pl/']);
+
+        $offerId = $response->json('data.id');
+
+        $updateResposne = $this->deleteJson(route('offers.destroy', $offerId))
+            ->assertStatus(204);
+
+
+        $this->assertDatabaseCount('offers', 0);
+    }
+
+    /** @test */
+    public function user_cannot_delete_other_user_offer(): void
+    {
+        $user = User::factory()->create();
+        $this->actingAs($user);
+
+        $offer = [
+            'url' => 'https://allegro.pl/',
+            'name' => 'New offer',
+            'is_active' => 1
+        ];
+
+        $response = $this->postJson(route('offers.store'), $offer)
+            ->assertCreated()
+            ->assertJsonFragment(['url' => 'https://allegro.pl/']);
+
+        $offerId = $response->json('data.id');
+
+        $user2 = User::factory()->create();
+        $this->actingAs($user2);
+
+        $updateResposne = $this->deleteJson(route('offers.destroy', $offerId))
+            ->assertStatus(404);
+
+
+        $this->assertDatabaseCount('offers', 1);
+    }
+
+    /** @test */
+    public function user_can_see_his_offers(): void
+    {
+        $user = User::factory()->create();
+        $this->actingAs($user);
+        $offersCount = 10;
+        $offers = Offer::factory()
+            ->count($offersCount)
+            ->create([
+                'user_id' => $user->id,
+            ]);
+
+        $response = $this->getJson(route('offers.index'))
+            ->assertOk()
+            ->assertJsonCount($offersCount, 'data');
+
+        foreach ($offers as $offer) {
+            $response->assertJsonFragment($offer->toArray());
+        }
+    }
+
+    /** @test */
+    public function user_can_see_list_of_his_offers_only(): void
```

Comment:
Czy ten test nie wystarczy, bo jeśli jest taki to ten poprzedni według mnie jest zbędny

Reply:
faktycznie :)